### PR TITLE
Build Subnet DHCPConfig with new NSX VPC 2.0 API

### PR DESCRIPTION
**Pull request by lxiaopei**
Based on new NSX VPC 2.0 API, we need to use below request body to create NSX subnet: {
        "id": "subnet-1",
        "resource_type": "VpcSubnet",
        "display_name": "Subnet 1",
        "description": "This is test VpcSubnet",
        "access_mode": "Private",
        "ipv4_subnet_size":64,
        "dhcp_config": {
            "enable_dhcp": false
        },
        "_revision": 0
    }

### CODE REVIEW

File: pkg/nsx/services/subnet/builder.go [hunk 1]
```diff
@@ -103,14 +103,6 @@ func (service *SubnetService) buildDHCPConfig(enableDHCP bool, poolSize int64) *
 	dhcpConfig := &model.VpcSubnetDhcpConfig{
 		EnableDhcp: Bool(enableDHCP),
 	}
-	if !enableDHCP {
```
Comment (timdengyun) [inline pkg/nsx/services/subnet/builder.go:106]: We don't need poolSize any more, why not remove this parameter?

**Review by timdengyun — APPROVED** [review]
LGTM

**Review by zhengxiexie — APPROVED** [review]
LGTM

**Pull request merged**